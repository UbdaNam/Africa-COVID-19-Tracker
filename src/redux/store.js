import { configureStore } from "@reduxjs/toolkit";
import countriesReducer from "./countries/countriesSlice";
import detailsReducer from "./details/detailsSlice";

const store = configureStore({
  reducer: {
    countries: countriesReducer,
    countryDetails: detailsReducer,
  },
});

export default store;
