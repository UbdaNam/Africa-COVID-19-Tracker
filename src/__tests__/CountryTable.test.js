import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import store from '../redux/store';
import CountryTable from '../components/DetailsTable';

const countryDetails = {
  countryDetail: {
    population: 15243234,
    cases: 458234,
    active: 2423,
    critical: 5,
    recovered: 234552,
    deaths: 98,
    tests: 9834573,
  },
};

const CountryTableSnap = renderer.create(
  <Provider store={store}>
    <Router>
      <CountryTable countryDetail={countryDetails.countryDetail} />
    </Router>
  </Provider>,
);

describe('Country table component render correctly', () => {
  test('Check if country table renders correctly', () => {
    expect(CountryTableSnap).toMatchSnapshot();
  });

  test("Check if text 'Population and Population count' exist inside the DOM", () => {
    render(
      <Provider store={store}>
        <Router>
          <CountryTable countryDetail={countryDetails.countryDetail} />
        </Router>
      </Provider>,
    );

    expect(screen.getByText('Population')).toBeInTheDocument();
    expect(screen.getByText('15,243,234')).toBeInTheDocument();
  });
});
