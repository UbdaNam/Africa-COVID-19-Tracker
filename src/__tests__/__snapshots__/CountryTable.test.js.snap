// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country table component render correctly Check if country table renders correctly 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        Population
      </td>
      <td>
        15,243,234
      </td>
    </tr>
    <tr>
      <td>
        Cases
      </td>
      <td>
        458,234
      </td>
    </tr>
    <tr>
      <td>
        Active
      </td>
      <td>
        2,423
      </td>
    </tr>
    <tr>
      <td>
        Critical
      </td>
      <td>
        5
      </td>
    </tr>
    <tr>
      <td>
        Recovered
      </td>
      <td>
        234,552
      </td>
    </tr>
    <tr>
      <td>
        Deaths
      </td>
      <td>
        98
      </td>
    </tr>
    <tr>
      <td>
        Tests
      </td>
      <td>
        9,834,573
      </td>
    </tr>
  </tbody>
</table>
`;
